libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2800,27 +2800,24 @@ string YulUtilFunctions::prepareStoreFunction(Type const& _type)
 
 string YulUtilFunctions::allocationFunction()
 {
-	string functionName = "allocateMemory";
+	string functionName = "allocate_memory";
 	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
 			function <functionName>(size) -> memPtr {
-				memPtr := mload(<freeMemoryPointer>)
-				let newFreePtr := add(memPtr, size)
-				// protect against overflow
-				if or(gt(newFreePtr, 0xffffffffffffffff), lt(newFreePtr, memPtr)) { <panic>() }
-				mstore(<freeMemoryPointer>, newFreePtr)
+				memPtr := <allocateUnbounded>()
```

Comment:
The code above uses `mem`. Should we just use `memPtr` consistently?

Reply:
Whereis 'mem' used?

Reply:
In the case above this.

Reply:
Also found `mpos` being used in places, like `YulUtilFunctions::externalCodeFunction`.

Reply:
> Also found mpos being used in places, like YulUtilFunctions::externalCodeFunction.

I think this was not addressed yet.

Reply:
Sorry for that, most of other comments were addressed so I went over this one too quickly and only checked for `mem` :)

Reply:
Can we keep unrelated code out of context of the "cleanup allocation functions" PR?

Reply:
Sure, you started to rename them, didn't you? :)

If there's renaming, do it properly in one sweep. I don't care too much about the local variables names, only about the allocate/finalize names to be not misleading.

Reply:
The stack element of a variable of memory reference type is called `mpos`, so I don't think it makes sense to rename all local variables called `mpos`.